### ISSUE TITLE
panic: assignment to entry in nil map

### ISSUE THREAD

**Issue by jjriv** [rclone/rclone#3652]
#### What is the problem you are having with rclone?
rclone copy is failing with the following message:
panic: assignment to entry in nil map

I should add that this does not happen with rclone version 1.49.5. It only happens in the beta.

#### What is your rclone version (output from `rclone version`)
```
rclone v1.49.5-244-g18c029e0-beta
- os/arch: linux/amd64
- go version: go1.13.3
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
CentOS Linux release 7.7.1908 (Core)
```
Linux 3.10.0-957.27.2.el7.x86_64 #1 SMP Mon Jul 29 17:46:05 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

####  Which cloud storage system are you using? (eg Google Drive)
Swift Object Storage and IBM Cloud Object Storage

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
```
rclone -vv --checkers 64 --transfers 64 --cache-db-purge --log-file rclone.log copy SWIFT:documents COS:cos-intervals-documents-1.1
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```2019/10/22 21:34:18 DEBUG : 12436/4/4fda3340e6050_JollyTrConfUs agnda Y12.pdf.enc: Size and modification time the same (differ by 0s, within tolerance 1ns)
2019/10/22 21:34:18 DEBUG : 12436/4/4fda3340e6050_JollyTrConfUs agnda Y12.pdf.enc: Unchanged skipping
panic: assignment to entry in nil map

goroutine 62 [running]:
github.com/rclone/rclone/backend/s3.(*Object).SetModTime(0xc0030aab60, 0x19796a0, 0xc000086300, 0x0, 0xed5417f86, 0x0, 0x1, 0x1)
        /home/runner/work/rclone/src/github.com/rclone/rclone/backend/s3/s3.go:1919 +0x111
github.com/rclone/rclone/fs/operations.equal(0x19796a0, 0xc000086300, 0x7f8709c19c30, 0xc003081800, 0x198f220, 0xc0030aab60, 0xc000010000, 0x23f0ea0)
        /home/runner/work/rclone/src/github.com/rclone/rclone/fs/operations/operations.go:221 +0x63a
github.com/rclone/rclone/fs/operations.Equal(...)
        /home/runner/work/rclone/src/github.com/rclone/rclone/fs/operations/operations.go:117
github.com/rclone/rclone/fs/operations.NeedTransfer(0x19796a0, 0xc000086300, 0x198f220, 0xc0030aab60, 0x198f520, 0xc003081800, 0x198f520)
        /home/runner/work/rclone/src/github.com/rclone/rclone/fs/operations/operations.go:1564 +0x795
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairChecker(0xc00045f340, 0xc000086240, 0xc000086280, 0xc00045f400)
        /home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:226 +0x510
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startCheckers
        /home/runner/work/rclone/src/github.com/rclone/rclone/fs/sync/sync.go:307 +0x9a
```

**Comment by ncw** [comment]
This looks like a duplicate of #3651 which was opened 4 hours before this!

Here is what I wrote:

This looks like that for some reason the SDK returned nil metadata on an object.

Can you try this

https://beta.rclone.org/branch/v1.49.5-245-gcf91aea5-fix-3651-s3-beta/ (uploaded in 15-30 mins)

**Comment by jjriv** [comment]
I've downloaded the beta and I'm running it now. It appears to be working, as I have yet to receive the error message. 

Your theory about the SDK returning nil metadata on an object makes sense. I've been receiving an error about metadata when trying to retrieve some objects:

```
rclone -vv lsl SWIFT:documents/10215/1/ | grep Tree_Master
2019/10/22 13:50:21 DEBUG : rclone: Version "v1.49.5-221-gf9f9d502-beta" starting with parameters ["rclone" "-vv" "lsl" "SWIFT:documents/10215/1/"]
2019/10/22 13:50:21 DEBUG : Using config file from "/root/.rclone.conf"
2019/10/22 13:55:09 DEBUG : 53872cdde57de_WM_Tree_Master-afa.pdf.enc: Failed to read metadata: object not found
   721388 2014-07-28 17:54:21.000000000 53872cdde57de_WM_Tree_Master-afa.pdf.enc
```
I don't suppose you have any tips on how I can copy objects that don't have their metadata counterpart?

**Comment by jjriv** [comment]
I can confirm that this is working now. I was able to copy the objects without error. As for my second comment about failed metadata... IBM has fixed that issue. Feel free to close this issue.

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.50

I also wrote a reply about the metadata which seems to have got lost, but it basically said that it was a problem with the origin server not having the object for some reason!
